### PR TITLE
Add managed environments to command contexts in preparation for managed identity work

### PR DESCRIPTION
**Pull request by MicroFish91**
The managed identity connection flow will rely on granting `acrPull` access to the parent container app environment.  In preparation for this upcoming work, this PR introduces a required environment context to extend from, forcing major commands to set managed environments before starting each flow.

I considered including this work as part of the next PR for which it is most relevant, however, most of this work is just reworking existing context types and setting managed environments on command contexts... so I figured it would be a lot easier to review separately since this is mostly distracting side-work supporting the main implementation.

### CODE REVIEW

File: src/commands/ManagedEnvironmentContext.ts [hunk 1]
```diff
@@ -0,0 +1,13 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import { type ManagedEnvironment } from "@azure/arm-appcontainers";
+import { type ISubscriptionActionContext } from "@microsoft/vscode-azext-utils";
+import { type AzureSubscription } from "@microsoft/vscode-azureresources-api";
+
+export interface RequiredManagedEnvironmentContext extends ISubscriptionActionContext {
```
Comment (MicroFish91) [inline src/commands/ManagedEnvironmentContext.ts:10]: I used the name "Required" here to help quickly identify that the managed environment resource is not optional.

File: src/commands/registryCredentials/identity/ManagedIdentityRegistryCredentialsContext.ts [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import { type RegistryCredentials } from "@azure/arm-appcontainers";
+import { type IContainerAppContext } from "../../IContainerAppContext";
+import { type CreateAcrContext } from "../../image/imageSource/containerRegistry/acr/createAcr/CreateAcrContext";
+import { type RequiredManagedEnvironmentContext } from "../../ManagedEnvironmentContext";
+
+export interface ManagedIdentityRegistryCredentialsContext extends CreateAcrContext, RequiredManagedEnvironmentContext, IContainerAppContext {
```
Comment (MicroFish91) [inline src/commands/registryCredentials/identity/ManagedIdentityRegistryCredentialsContext.ts:11]: This flow has not technically been introduced yet, but I went ahead and set up the context shape so we can enforce the new type requirements.

File: src/commands/ManagedEnvironmentContext.ts (hunk 1 above)
Comment (nturinski) [reply src/commands/ManagedEnvironmentContext.ts:10]: Discussed offline, but I think we settled on using a naming convention that would put the resource ala `ManagedEnvironment` first.

File: src/commands/createContainerApp/CreateContainerAppContext.ts [hunk 3]
```diff
@@ -14,9 +13,6 @@ import { type IngressBaseContext } from '../ingress/IngressContext';
 
 export interface CreateContainerAppBaseContext extends IResourceGroupWizardContext, ImageSourceBaseContext, IngressBaseContext, IContainerAppContext, ExecuteActivityContext {
```
Comment (nturinski) [inline src/commands/createContainerApp/CreateContainerAppContext.ts:14]: `ContainerAppCreateBaseContext`?

File: src/commands/createManagedEnvironment/CreateManagedEnvironmentContext.ts [hunk 4]
```diff
@@ -9,13 +9,13 @@ import { type IResourceGroupWizardContext } from '@microsoft/vscode-azext-azureu
 import { type ExecuteActivityContext, type ISubscriptionActionContext } from '@microsoft/vscode-azext-utils';
 import { type AzureSubscription } from "@microsoft/vscode-azureresources-api";
 
-export interface IManagedEnvironmentContext extends ISubscriptionActionContext, IResourceGroupWizardContext, ExecuteActivityContext {
+export interface CreateManagedEnvironmentContext extends ISubscriptionActionContext, IResourceGroupWizardContext, ExecuteActivityContext {
```
Comment (nturinski) [inline src/commands/createManagedEnvironment/CreateManagedEnvironmentContext.ts:12]: `ManagedEnvironmentCreateContext`?

**Comment by MicroFish91** [comment]
Updated with feedback, this should be ready for review again

**Review by nturinski — APPROVED** [review]
LGTM

**Pull request merged**